src/snyk/common/api/apiСlient.ts

```diff
@@ -35,15 +36,32 @@ export class SnykApiClient implements ISnykApiClient {
 
   get<T = unknown, R = AxiosResponse<T>>(url: string, config?: AxiosRequestConfig): Promise<R> {
     this.http.interceptors.request.use(req => {
-      req.baseURL = `${this.configuration.authHost}/api/v1/`;
-      req.headers = {
-        ...req.headers,
-        Authorization: `token ${this.configuration.token}`,
-      } as { [header: string]: string };
+      if (req.method === 'get') {
+        req.baseURL = `${this.configuration.authHost}/api/v1/`;
+        req.headers = {
+          ...req.headers,
+          Authorization: `token ${this.configuration.token}`,
+        } as { [header: string]: string };
+      }
 
       return req;
     });
 
     return this.http.get<T, R>(url, config);
   }
+
+  post<T = unknown, R = AxiosResponse<T>>(url: string, data?: unknown, config?: AxiosRequestConfig): Promise<R> {
+    this.http.interceptors.request.use(req => {
+      if (req.method === 'post') {
+        req.baseURL = this.configuration.baseApiUrl;
```

Comment:
This class should work with the same base Url for API. If there's a need to work with different API url, you either create another class or introduce type parameter here indicating what type of Snyk Api we want to POST to.

Reply:
You want to create a new class just for a single method? I think it's a bit overkill and duplication of classes.

Reply:
It is important to follow single responsibility principle. Use another suggested option then, if you don't like another class. But first determine what's the difference between these two APIs and only then make a decision please.

Reply:
I understand and I agree on that, it looks like a duplication now that I look at it. But I think this is not the case because apiClient is common and generic. 
Another solution can be that we'll use a helper function/class that will return the right url according to the request method (in this case). What do you think about it?

Reply:
If you can keep single responsibility principle with a new function/class, then sure :) Sometimes it's better to duplicate rather than making a thing less comprehensible when glancing at the code.